docs/source/development/documentation.rst

```diff
@@ -1,3 +1,127 @@
+	        If size mismatches detected or input matrices are not square.
+
+	    Notes
+	    -----
+	    For a specific class of priors the probabilistic linear solver recovers the iterates of the conjugate gradient
+
+	    ...
+
+	    References
+	    ----------
+	    .. [1] Wenger, J. and Hennig, P., Probabilistic Linear Solvers for Machine Learning, 2020
+
+	    ...
+
+	    See Also
+	    --------
+	    bayescg : Solve linear systems with prior information on the solution.
+
+	    Examples
+	    --------
+	    >>> import numpy as np
+	    >>> np.random.seed(1)
+	    >>> n = 20
+	    >>> A = np.random.rand(n, n)
+	    >>> A = 0.5 * (A + A.T) + 5 * np.eye(n)
+	    >>> b = np.random.rand(n)
+	    >>> x, A, Ainv, info = problinsolve(A=A, b=b)
+	    >>> print(info["iter"])
+	    10
+	    """
+
+
+**General Rules**
+
+- Cover :code:`Parameters`, :code:`Returns`, :code:`Raises` and :code:`Examples` at every publicly visible docstring---in that order.
+- Examples are tested via doctest. Ensure :code:`doctest` does not fail.
+- Include appropriate references, in particular for new probabilistic numerical methods.
+- Do not use docstrings as a clutch for spaghetti code.
+
+**Parameters**
+
+- Hyperparameters should have default values and explanations on how to choose them.
+- Array-type inputs (:code:`np.ndarray`, :code:`csr_matrix`, ...) are documented as :code:`array_like` (in accordance with NumPy).
+- Provide shape hints via :code:`foo : array_like, shape=(a, b)` wherever possible.
+- Callables are used as parameters if the expected signature is part of the docstring: :code:`foobar : callable, signature=(t, \*\*kwargs)`. Backslashes remove semantic meaning from special characters.
+
+**Style**
+
+- Stick to the imperative style of writing in the docstring header (i.e.: first line).
+
+  - Yes: "Compute the value". 
+  - No: "This function computes the value / Let's compute the value".
+  
+  The rest of the explanation talks about the function, e. g. "This function computes the value by computing another value".
+- Use full sentences inside docstrings when describing something.
+
+  - Yes: "This value is irrelevant, because it is not being passed on"
+  - No: "Value irrelevant, not passed on". 
+- When in doubt, more explanation rather than less. A little text inside an example can be helpful, too.
+- A little maths can go a long way, but too much usually adds confusion.
+
+Interface Documentation
+************************
+
+Which functions and classes actually show up in the documentation is determined by an :code:`__all__` statement in the 
+corresponding :code:`__init__.py` file inside a module. The order of this list is also reflected in the documentation. 
+For example, :code:`linalg` has the following :code:`__init__.py`:
+
+.. code-block:: python
+
+	"""
+	Linear Algebra.
+
+	This package implements common operations and (probabilistic) numerical methods for linear algebra.
+	"""
+
+	from probnum.linalg.linearsolvers import *
```

Comment:
I guess this concludes that we indeed do not want to check for PEP-8 errors regarding star imports in `__ini__.py` files

Reply:
I forgot to change this. Based on the result of our discussion explicit imports need to be specified in the `__all__` variable anyway, therefore making explicit imports little additional work.

Reply:
I agree that in general we should avoid star imports. But for some reason when I looked into fixing those issues in `__init__.py` files I came to the conclusion that there might also be reasons for having an exception to this rule in those files. 

Anyways, thanks for fixing it in the documentation, this way at least we don't explicitly encourage contributors to use them :)